eppo_client/configuration_requestor.py

```diff
@@ -2,44 +2,75 @@
-from eppo_client.models import Flag
+from eppo_client.models import BanditData, Flag
 
 logger = logging.getLogger(__name__)
 
 
 UFC_ENDPOINT = "/flag-config/v1/config"
+BANDIT_ENDPOINT = "/flag-config/v1/bandits"
 
 
 class ExperimentConfigurationRequestor:
     def __init__(
         self,
         http_client: HttpClient,
-        config_store: ConfigurationStore[Flag],
+        flag_config_store: ConfigurationStore[Flag],
+        bandit_config_store: ConfigurationStore[BanditData],
     ):
         self.__http_client = http_client
-        self.__config_store = config_store
+        self.__flag_config_store = flag_config_store
+        self.__bandit_config_store = bandit_config_store
         self.__is_initialized = False
 
     def get_configuration(self, flag_key: str) -> Optional[Flag]:
         if self.__http_client.is_unauthorized():
             raise ValueError("Unauthorized: please check your API key")
-        return self.__config_store.get_configuration(flag_key)
+        return self.__flag_config_store.get_configuration(flag_key)
+
+    def get_bandit_model(self, bandit_key: str) -> Optional[BanditData]:
+        if self.__http_client.is_unauthorized():
+            raise ValueError("Unauthorized: please check your API key")
+        return self.__bandit_config_store.get_configuration(bandit_key)
 
     def get_flag_keys(self):
-        return self.__config_store.get_keys()
+        return self.__flag_config_store.get_keys()
+
+    def get_bandit_keys(self):
+        return self.__bandit_config_store.get_keys()
+
+    def fetch_flags(self):
+        return self.__http_client.get(UFC_ENDPOINT)
 
-    def fetch_and_store_configurations(self) -> Dict[str, Flag]:
+    def fetch_bandits(self):
+        return self.__http_client.get(BANDIT_ENDPOINT)
+
+    def store_flags(self, flag_data) -> Dict[str, Flag]:
+        flag_config_dict = cast(dict, flag_data.get("flags", {}))
+        flag_configs = {key: Flag(**config) for key, config in flag_config_dict.items()}
+        self.__flag_config_store.set_configurations(flag_configs)
+        return flag_configs
+
+    def store_bandits(self, bandit_data) -> Dict[str, BanditData]:
+        bandit_configs = {
+            config["banditKey"]: BanditData(**config)
+            for config in cast(dict, bandit_data.get("bandits", []))
+        }
+        self.__bandit_config_store.set_configurations(bandit_configs)
+        return bandit_configs
+
+    def fetch_and_store_configurations(self):
         try:
-            configs_dict = cast(
-                dict, self.__http_client.get(UFC_ENDPOINT).get("flags", {})
-            )
-            configs = {key: Flag(**config) for key, config in configs_dict.items()}
-            self.__config_store.set_configurations(configs)
+            flag_data = self.fetch_flags()
+            self.store_flags(flag_data)
+
+            if flag_data.get("bandits", {}):
+                bandit_data = self.fetch_bandits()
```

Comment:
Side note, one of the fast/future follows is to decouple bandit polling frequency from configuration polling. That way anybody that is using bandits doesn't see their costs immediately double. ([slack thread](https://eppo-group.slack.com/archives/C046TEQ89K7/p1712270378395689))

Reply:
That's fair -- another option is to have bandits only ship with server SDK requests using API key permissions